### ISSUE TITLE
Query - regarding missing messages

### ISSUE THREAD

**Issue by asimpatnaik** [liftbridge-io/go-liftbridge#9]
Hi,
Below is the setup:
1. 2 subscribers subscribe to the same subject and the same stream identifier (Name attribute of the stream)
2. 1st subscriber is subscribing with option StartAtOffset and the 2nd subscriber is subscribing with default option
Scenario 1 - subscribers are on different boxes:
Both capture msgs and process them without any issues

Scenario 2 - subscribers are on the same box:
Some msgs supposed to be picked up by the 2nd subscriber are not picked

It will be great if you explain this anomaly and suggest a solution. Do let me know if there is something silly that i might have done which is a possibility :)

Thanks a lot.

**Comment by asimpatnaik** [comment]
Thanks for the reply.
Yes, messages are being published to the same subject. 2 subscribers are
listening on the same subject and same stream name. The issue of message
loss is happening when both subscribers are running on the same machine.

**Comment by asimpatnaik** [comment]
My apologies first and thanks a lot for your valuable time. The prognosis of the issue wasn't right. It wasn't related to loss of messages, which we discovered later with more diagnostic msgs logging. We figured out that there was an inter network communication when two different machines were involved and as a result, the time delay of receiving the publication was good enough for the db to persist the data which later was being read using the published msg. With the machines being the same, the pub-sub was faster than the db commit time and as a result, no data was available when being read. There's absolutely no issue with lift-bridge.
